packages/single-selection-mixin/single-selection-mixin.ts

```diff
@@ -0,0 +1,91 @@
+import { property, PropertyValues } from 'lit-element';
+import { DisabledStateInterface } from '@vaadin/disabled-state-mixin';
+import { KeyboardDirectionInterface } from '@vaadin/keyboard-direction-mixin';
+import { KeyboardDirectionClass } from '@vaadin/keyboard-direction-mixin/keyboard-direction-class';
+import { SelectedStateClass } from '@vaadin/selected-state-mixin/selected-state-class';
+import { SelectedStateInterface } from '@vaadin/selected-state-mixin';
+import { SlottedItemsInterface } from '@vaadin/slotted-items-mixin';
+import { SlottedItemsClass } from '@vaadin/slotted-items-mixin/slotted-items-class';
+import { SingleSelectionClass } from './single-selection-class';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type Constructor<T = object> = new (...args: any[]) => T;
+
+export interface SingleSelectionInterface {
+  selected: number | null | undefined;
+}
+
+const filterItems = (nodes: Node[]) => {
+  return nodes.filter((node): node is SelectedStateClass => {
+    return (
+      node.nodeType === Node.ELEMENT_NODE &&
+      (node.constructor as typeof SelectedStateClass).hasSelectedStateMixin
+    );
+  });
+};
+
+export const SingleSelectionMixin = <
+  T extends Constructor<
+    KeyboardDirectionInterface &
+      KeyboardDirectionClass &
+      SlottedItemsInterface &
+      SlottedItemsClass &
+      SingleSelectionClass
+  >
+>(
+  base: T
+): T & Constructor<SingleSelectionInterface> => {
+  class SingleSelection extends base {
+    /**
+     * The index of the item selected in the items array.
+     */
+    @property({ type: Number, reflect: true }) selected: number | null | undefined;
+
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      this.addEventListener('click', event => {
+        this._onClick(event);
+      });
+
+      this.addEventListener('keyup', event => {
+        this._onKeyUp(event);
+      });
+    }
+
+    protected updated(props: PropertyValues) {
+      super.updated(props);
+
+      if (props.has('selected')) {
+        const { items, selected } = this;
+        items.forEach((item, idx) => {
+          // eslint-disable-next-line no-param-reassign
+          (item as SelectedStateClass & SelectedStateInterface).selected = idx === selected;
+        });
+      }
+    }
+
+    protected _onClick(event: MouseEvent) {
+      this._setSelected(event);
+    }
+
+    protected _onKeyUp(event: KeyboardEvent) {
+      this._setSelected(event);
+    }
+
+    private _setSelected(event: MouseEvent | KeyboardEvent) {
+      if (event.metaKey || event.shiftKey || event.ctrlKey) {
+        return;
+      }
+
+      const path = event.composedPath() as Node[];
+      const item = filterItems(path)[0] as SelectedStateClass & DisabledStateInterface;
```

Comment:
I think it could be helpful (feature) to save a reference to this `item` as a publicly accessible property e.g. `selectedElement`.

Possibly also having API to set the selected item by element reference (instead of by index).

Or should we open a new feature issue about this?

What do you think?

Reply:
Submitted https://github.com/vaadin/component-mixins/issues/24